src/errors.rs

```diff
@@ -6,6 +6,7 @@ use thiserror::Error;
 pub(crate) type GpxResult<T> = Result<T, GpxError>;
 
 #[derive(Error, Debug)]
+#[non_exhaustive]
```

Comment:
FWIW, this is also a breaking change (code that used to match on `GpxError` will no longer compile).

Reply:
Yep.. I don't know what to do with this. Adding to enum also breaking change. If some one match all case

Reply:
I'm curious why you made this non_exhaustive?

Is it to avoid breaking changes in the future?  I could see why that would make sense for a public error type like this.

Reply:
Yes this can avoid breaking changes in the future